android/src/com/mapswithme/maps/location/TrackRecorderWakeReceiver.java

```diff
@@ -23,6 +23,11 @@ public void onReceive(Context context, Intent intent)
                  + !backgroundTracker(context).isForeground();
     LOGGER.i(TAG, msg);
     CrashlyticsUtils.INSTANCE.log(Log.INFO, TAG, msg);
-    TrackRecorder.INSTANCE.onWakeAlarm();
+    try {
+      TrackRecorder.INSTANCE.onWakeAlarm();
+    }catch(NullPointerException e){
```

Comment:
Catching NPE is almost always a bad idea. In this case, it means that we a trying to use `TrackRecorder` before it is initialized. But we cannot initialize only just it because it can depend on other singletons that are also still not initialized.

I'll think about how to fix it properly.